src/lib/watcher/runner.ts

```diff
@@ -126,25 +119,51 @@ register({
     IPC.client.senders.moduleImported({ filePath })
   })
 
-  if (!process.env.NEXUS_EVAL) {
-    throw new Error('process.env.NEXUS_EVAL is required')
-  }
+  const startModuleFileName = layout.sourceRoot + '/index.ts'
+  const startModule = tsNodeRegister.compile(
+    createStartModuleContent({
+      internalStage: 'dev',
+      layout: layout,
+      inlineSchemaModuleImports: true,
+    }),
+    startModuleFileName
+  )
 
-  evalScript(process.env.NEXUS_EVAL)
-
-  function evalScript(script: string) {
-    const EVAL_FILENAME = process.env.NEXUS_EVAL_FILENAME!
-    const module = new Module(EVAL_FILENAME)
-    module.filename = EVAL_FILENAME
-    module.paths = (Module as any)._nodeModulePaths(cwd)
-    ;(global as any).__filename = EVAL_FILENAME
-    ;(global as any).__dirname = cwd
-    ;(global as any).exports = module.exports
-    ;(global as any).module = module
-    ;(global as any).require = module.require.bind(module)
-
-    new Script(script, {
-      filename: EVAL_FILENAME,
-    }).runInThisContext()
-  }
-})()
```

Comment:
What was the reason for inlining everything below? Less indirections? Given how meta this stuff is, I felt like it was better contained in a single function

Reply:
Yeah it was about removing indirection. I see your point! Agree